src/parsers/pm2ps_parser.jl

```diff
@@ -263,41 +263,76 @@ function make_thermal_gen(gen_name, d, bus)
     return thermal_gen
 end
 
-function read_gen(data,Buses)
+"""
+Transfer generators to ps_dict according to their classification
+"""
+function read_gen(data, Buses, genmap_file=nothing)
+
+    if genmap_file == nothing
+        genmap_file = joinpath(dirname(dirname(pathof(PowerSystems))),
```

Comment:
Would it be better to make a global constant for the filename and then have that value be the default?

Reply:
I tried that first, but I couldn't figure out how to get the default path correctly.

Reply:
I've left this logic. Please also check how I implemented passing this new argument through from parsestandardfiles. I couldn't figure out how to use the existing mulitple dispatch implementation with default arguments and keywords, and so I made it just one method. Feel free to change this if you know of something more elegant.

Reply:
Now changed to look for existence of keyword argument.